source/chapters/appendix/mixxx_controls.rst

```diff
@@ -99,6 +99,92 @@ To use ``*_toggle`` the respective shortcut for scripts is:
    script.toggleControl(group, "keylock_toggle", 100);
 
 
+The ``[App]`` group
+~~~~~~~~~~~~~~~~~~~
+
+The :mixxx:cogroupref:`[App]` group contains controls that do not belong to a specific channel, the mixer or the effects engine.
+
+.. mixxx:control:: [App],indicator_250ms
+
+   Alternates between 0.0 and 1.0 every 250 milliseconds.
+
+   This control may be used to implement a blinking LED in JavaScript and is
+   guaranteed to light up at the same time as
+   :mixxx:coref:`[ChannelN],cue_indicator` and
+   :mixxx:coref:`[ChannelN],play_indicator` when these are blinking (depending
+   on the currently chosen :ref:`cue mode <interface-cue-modes>`).
+
+   :range: binary, read-only
+   :feedback: None
+
+   .. versionadded:: 2.4.0
+
+.. mixxx:control:: [App],indicator_500ms
+
+   Alternates between 0.0 and 1.0 every 500 milliseconds.
+
+   This control may be used to implement a blinking LED in JavaScript and is
+   guaranteed to light up at the same time as
+   :mixxx:coref:`[ChannelN],cue_indicator` and
+   :mixxx:coref:`[ChannelN],play_indicator` when these are blinking (depending
+   on the currently chosen :ref:`cue mode <interface-cue-modes>`).
+
+   :range: binary, read-only
+   :feedback: None
+
+   .. versionadded:: 2.4.0
+
+
+.. mixxx:control:: [App],num_decks
+
+   The number of decks currently enabled.
+
+   :range: integer
+   :feedback: None
+
+   .. versionadded:: 2.4.0
+
+
+.. mixxx:control:: [App],num_samplers
+
+   The number of samplers currently enabled.
```

Comment:
Does this value change when switching the skin? I think a mapping developer needs to know, if it's constant over the runtime of the mapping.

Reply:
I think it's constant, but I don't really know. I did not change the current description, just moved it into a new `[App]` section. If you have a new wording please make a suggestion and I'll include it. Otherwise this could be improved in a later PR (maybe file an issue).

Reply:
IIRC the CO is adjusted to the maximum of
* num_samplers set in skin manifest
* samplers.xml loaded at startup
* any other sampler bank XML loaded manually later on